lib/api/api.js

```diff
@@ -184,14 +186,107 @@ const api = {
             }
             return { returnTagCount, isImplicitDeny };
         }
+        let formDataParser;
+        let fileEventData = null;
+
+        if (apiMethod === 'objectPost') {
+            formDataParser = busboy({ headers: request.headers });
+        }
 
         return async.waterfall([
+            next => {
+                if (apiMethod === 'objectPut' || apiMethod === 'objectPutPart') {
+                    return next(null);
+                }
+                if (apiMethod === 'objectPost' && request.headers['content-type'].includes('multipart/form-data')) {
+                    writeContinue(request, response);
+
+                    request.formData = {};
+                    let totalFieldSize = 0;
+                    const MAX_FIELD_SIZE = 20 * 1024; // 20KB
+                    formDataParser.on('field', (fieldname, val) => {
+                        totalFieldSize += Buffer.byteLength(val, 'utf8');
+                        if (totalFieldSize > MAX_FIELD_SIZE) {
+                            const err = errors.MaxPostPreDataLengthExceeded;
+                            formDataParser.emit('error', err);
+                        }
+                        // Convert fieldname to lowercase for case-insensitive comparison
+                        const lowerFieldname = fieldname.toLowerCase();
+                        request.formData[lowerFieldname] = val;
+                    });
+
+                    formDataParser.on('file', (fieldname, file, filename, encoding, mimetype) => {
+                        fileEventData = { fieldname, file, filename, encoding, mimetype };
+                        // no key field == error
+                        if ('key' in request.formData) {
+                            return next(null);
+                        }
```

Comment:
key also needs to be validated, you can't save an empty key

```
    <Code>InvalidArgument</Code>
    <Message>User key must have a length greater than 0.</Message>
```
Also it has a max size, but this one is not validated by cloudserver on PUT anyways, but PUT will have 414 Request-URI Too Long, however here as we are not in uri, it will trigger other error related to form-data field, so it's better to validate it.
```
    <Code>KeyTooLongError</Code>
    <Message>Your key is too long</Message>
    <Size>1025</Size>
    <MaxSizeAllowed>1024</MaxSizeAllowed>
```

Also you can't have multiple times the same key field
```
    <Code>InvalidArgument</Code>
    <Message>POST only supports one key parameter per request.</Message>
    <ArgumentName>key</ArgumentName>
    <ArgumentValue>2</ArgumentValue>
```

Reply:
Implemented
https://github.com/scality/cloudserver/pull/5601/files#diff-69284c3576c9c4651c9ab95928cbb676d75009ef4ee5becb238b45e22bb20efeR50
Tested
https://github.com/scality/cloudserver/pull/5601/files#diff-69284c3576c9c4651c9ab95928cbb676d75009ef4ee5becb238b45e22bb20efeR143

Same for key too long
https://github.com/scality/cloudserver/pull/5601/files#diff-69284c3576c9c4651c9ab95928cbb676d75009ef4ee5becb238b45e22bb20efeR46
(test)
https://github.com/scality/cloudserver/pull/5601/files#diff-406efa736d8942ad5f5ba3c3404803f3748ef0d0cd44a20b48f5ce074c3105cdR637

and multiple keys
https://github.com/scality/cloudserver/pull/5601/files#diff-69284c3576c9c4651c9ab95928cbb676d75009ef4ee5becb238b45e22bb20efeR58
(test)
https://github.com/scality/cloudserver/pull/5601/files#diff-406efa736d8942ad5f5ba3c3404803f3748ef0d0cd44a20b48f5ce074c3105cdR367

Reply:
see https://github.com/scality/cloudserver/pull/5601#discussion_r1690995359 I'm closing this